src/storage/src/table/error.rs

```diff
@@ -14,9 +14,75 @@
 
 use std::backtrace::Backtrace;
 
-use risingwave_common::error::{ErrorCode, RwError};
 use thiserror::Error;
 
+#[derive(Error, Debug)]
+enum StorageTableErrorInner {
+    #[error("Table iterator Error {0}")]
+    TableIterator(String),
+
+    #[error("State store iterator error {0}")]
+    StateStoreIterator(String),
+
+    #[error("Wait epoch error {0}")]
+    WaitEpoch(String),
+
+    #[error("Deserialize row error {0}.")]
+    DeserializeRow(String),
+
+    #[error("state store get error {0}.")]
+    StateStoreGet(String),
+}
+
+#[derive(Error)]
+#[error("{inner}")]
+pub struct StorageTableError {
+    #[from]
+    inner: StorageTableErrorInner,
+    backtrace: Backtrace,
+}
+
+impl StorageTableError {
+    pub fn table_iterator_error(error: impl ToString) -> StorageTableError {
+        StorageTableErrorInner::TableIterator(error.to_string()).into()
```

Comment:
By converting the inner storage error (actually `HummockError`) to display string, I think we'll lose the backtrace on the hummock side. This may bring difficulties to our debugging.